File: bigdata/src/main/java/com/hl/bigdata/flink/stream/scala/source/MySourceNonParallelism.scala
package com.hl.bigdata.flink.stream.scala.source

import org.apache.flink.streaming.api.functions.source.SourceFunction
import org.apache.flink.streaming.api.scala._

/**
 * @author huanglin
 * @date 2024/03/12 15:32
 */
class MySourceNonParallelism extends SourceFunction[Int]{

  var sum: Int           = 0
  var isRunning: Boolean = true


  override def run(sourceContext: SourceFunction.SourceContext[Int]): Unit = {
    while(isRunning) {
      sourceContext.collect(sum)
      sum = sum + 1
      Thread.sleep(1000)
    }
  }

  override def cancel(): Unit = isRunning = false
}

File: bigdata/src/main/java/com/hl/bigdata/flink/stream/java/source/MySourceNonParallelism.java
package com.hl.bigdata.flink.stream.java.source;

import org.apache.flink.streaming.api.functions.source.SourceFunction;

/**
 * @author huanglin
 * @date 2024/03/08 18:20
 */
public class MySourceNonParallelism implements SourceFunction<Integer> {

    private int     count     = 1;
    private boolean isRunning = true;

    @Override
    public void run(SourceContext<Integer> sourceContext) throws Exception {
        while(isRunning) {
            sourceContext.collect(count);
            count++;
            Thread.sleep(1000);
        }
    }

    @Override
    public void cancel() {

    }
}
